src/contexts/Drive/Drive.context.tsx

```diff
@@ -46,10 +50,75 @@ interface DriveContextProviderProps {
 const logger = new BaseLogger({
   tag: 'DRIVE_CONTEXT',
 });
+
+export const getModifiedDriveItemsAndUpdateLocalCache = async () => {
+  const lastUpdatedAt = (await asyncStorageService.getItem(AsyncStorageKey.LastUpdatedAt)) ?? new Date().toISOString();
+
+  const [modifiedFiles, modifiedFolders] = await Promise.all([
+    driveFileService.getModifiedFiles({
+      updatedAt: lastUpdatedAt,
+      status: 'ALL',
+    }),
+    driveFolderService.getModifiedFolders({
+      updatedAt: lastUpdatedAt,
+      status: 'ALL',
+    }),
+  ]).catch((error) => {
+    errorService.reportError(error);
+    return [[], []];
+  });
+
+  if (!modifiedFiles || !modifiedFolders || (!modifiedFiles.length && !modifiedFolders.length)) return;
+
+  const modifiedFilesIds = modifiedFiles.map((file) => file.id);
+  const modifiedFoldersIds = modifiedFolders.map((folder) => folder.id);
+
+  modifiedFilesIds.forEach(async (fileId) => {
+    await driveLocalDB.deleteItem({ id: fileId });
+  });
+
+  modifiedFoldersIds.forEach(async (folderId) => {
+    await driveLocalDB.deleteFolderRecord(folderId);
+  });
+
+  // Get the last updatedAt date from the modified files and folders
+  let lastUpdatedAtFromModifiedFiles;
+  let lastUpdatedAtFromModifiedFolders;
+
+  if (modifiedFiles.length) lastUpdatedAtFromModifiedFiles = _.maxBy(modifiedFiles, 'updatedAt')?.updatedAt;
+
+  if (modifiedFolders.length)
+    lastUpdatedAtFromModifiedFolders = _.maxBy(modifiedFolders, 'updatedAt')?.updatedAt.toString();
+
+  // Get the most recent date
+  const newLastUpdatedAt = _.max([lastUpdatedAtFromModifiedFiles, lastUpdatedAtFromModifiedFolders]);
+
+  await asyncStorageService.saveItem(AsyncStorageKey.LastUpdatedAt, newLastUpdatedAt ?? new Date().toISOString());
+};
+
 export const DriveContextProvider: React.FC<DriveContextProviderProps> = ({ children, rootFolderId }) => {
   const [viewMode, setViewMode] = useState(DriveListViewMode.List);
   const [driveFoldersTree, setDriveFoldersTree] = useState<DriveFoldersTree>({});
   const [currentFolder, setCurrentFolder] = useState<FetchFolderContentResponseWithThumbnails | null>(null);
+  const currentFolderId = useRef<number | null>(null);
+
+  const handleAppStateChange = async (state: AppStateStatus) => {
+    if (state === 'active') {
+      if (currentFolderId.current)
+        loadFolderContent(currentFolderId.current, { pullFrom: ['network'] }).catch((error) => {
+          errorService.reportError(error);
+        });
+    }
+  };
+
+  useEffect(() => {
+    const listener = appService.onAppStateChange(handleAppStateChange);
```

Comment:
move this to a ref, this variable can get lost without being removed cause it only lives in this scope, if moving to a ref, you can make sure you only have one always